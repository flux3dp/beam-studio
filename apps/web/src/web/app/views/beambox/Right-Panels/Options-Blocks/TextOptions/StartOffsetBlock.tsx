import React from 'react';

import i18n from 'helpers/i18n';
import ObjectPanelItem from 'app/views/beambox/Right-Panels/ObjectPanelItem';
import styles from 'app/views/beambox/Right-Panels/OptionsPanel.module.scss';
import UnitInput from 'app/widgets/Unit-Input-v2';
import { useIsMobile } from 'helpers/system-helper';

interface Props {
  value: number;
  onValueChange: (val: number) => void;
}

export default function StartOffsetBlock({ value, onValueChange }: Props): JSX.Element {
  const LANG = i18n.lang.beambox.right_panel.object_panel.option_panel;
  const isMobile = useIsMobile();
  return isMobile ? (
    <ObjectPanelItem.Number
      id="start_offset"
      label={LANG.start_offset}
      value={value}
      min={0}
      max={100}
      updateValue={onValueChange}
      unit=""
      decimal={0}
    />
  ) : (
    <div className={styles['option-block']}>
      <div className={styles.label}>{LANG.start_offset}</div>
      <UnitInput
        min={0}
        max={100}
        unit=""
        decimal={0}
        className={{ [styles['option-input']]: true }}
        defaultValue={value}
        getValue={onValueChange}
      />
    </div>
  );
}
