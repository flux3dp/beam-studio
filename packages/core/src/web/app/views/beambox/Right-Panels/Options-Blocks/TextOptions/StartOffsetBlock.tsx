import React from 'react';

import ObjectPanelItem from '@core/app/views/beambox/Right-Panels/ObjectPanelItem';
import styles from '@core/app/views/beambox/Right-Panels/OptionsPanel.module.scss';
import UnitInput from '@core/app/widgets/Unit-Input-v2';
import i18n from '@core/helpers/i18n';
import { useIsMobile } from '@core/helpers/system-helper';

interface Props {
  onValueChange: (val: number) => void;
  value: number;
}

export default function StartOffsetBlock({ onValueChange, value }: Props): React.JSX.Element {
  const LANG = i18n.lang.beambox.right_panel.object_panel.option_panel;
  const isMobile = useIsMobile();

  return isMobile ? (
    <ObjectPanelItem.Number
      decimal={0}
      id="start_offset"
      label={LANG.start_offset}
      max={100}
      min={0}
      unit=""
      updateValue={onValueChange}
      value={value}
    />
  ) : (
    <div className={styles['option-block']}>
      <div className={styles.label}>{LANG.start_offset}</div>
      <UnitInput
        className={{ [styles['option-input']]: true }}
        decimal={0}
        defaultValue={value}
        getValue={onValueChange}
        max={100}
        min={0}
        unit=""
      />
    </div>
  );
}
