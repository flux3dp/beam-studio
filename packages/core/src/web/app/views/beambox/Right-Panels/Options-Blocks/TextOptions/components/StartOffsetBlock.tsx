import React from 'react';

import ObjectPanelItem from '@core/app/views/beambox/Right-Panels/ObjectPanelItem';
import styles from '@core/app/views/beambox/Right-Panels/OptionsPanel.module.scss';
import UnitInput from '@core/app/widgets/Unit-Input-v2';
import { useIsMobile } from '@core/helpers/system-helper';
import useI18n from '@core/helpers/useI18n';

interface Props {
  hasMultiValue?: boolean;
  onValueChange: (val: number) => void;
  value: number;
}

export default function StartOffsetBlock({ hasMultiValue, onValueChange, value }: Props): React.JSX.Element {
  const LANG = useI18n().beambox.right_panel.object_panel.option_panel;
  const isMobile = useIsMobile();

  return isMobile ? (
    <ObjectPanelItem.Number
      decimal={0}
      hasMultiValue={hasMultiValue}
      id="start_offset"
      label={LANG.start_offset}
      max={100}
      min={0}
      unit=""
      updateValue={onValueChange}
      value={value}
    />
  ) : (
    <div className={styles['option-block']}>
      <div className={styles.label}>{LANG.start_offset}</div>
      <UnitInput
        className={{ [styles['no-unit']]: true, [styles['option-input']]: true }}
        decimal={0}
        defaultValue={value}
        displayMultiValue={hasMultiValue}
        getValue={onValueChange}
        max={100}
        min={0}
        unit=""
      />
    </div>
  );
}
