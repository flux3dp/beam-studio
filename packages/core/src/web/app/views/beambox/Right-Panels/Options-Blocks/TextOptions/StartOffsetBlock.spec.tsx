import React from 'react';

import { fireEvent, render } from '@testing-library/react';

const useIsMobile = jest.fn();

jest.mock('@core/helpers/system-helper', () => ({
  useIsMobile: () => useIsMobile(),
}));

jest.mock('@core/helpers/i18n', () => ({
  lang: {
    beambox: {
      right_panel: {
        object_panel: {
          option_panel: {
            start_offset: 'Text Offset',
          },
        },
      },
    },
  },
}));

jest.mock('@core/app/widgets/Unit-Input-v2', () => ({ className, decimal, defaultValue, getValue, max, min }: any) => (
  <div>
    mock-unit-input min:{min}
    max:{max}
    decimal:{decimal}
    defaultValue:{defaultValue}
    className:{JSON.stringify(className)}
    <input className="unit-input" onChange={(e) => getValue(+e.target.value)} />
  </div>
));

jest.mock('@core/app/views/beambox/Right-Panels/ObjectPanelItem', () => ({
  Number: ({ decimal, id, label, max, min, updateValue, value }: any) => (
    <div>
      mock-number-item id:{id}
      label:{label}
      min:{min}
      max:{max}
      decimal:{decimal}
      value:{value}
      <input className="unit-input" onChange={(e) => updateValue(+e.target.value)} />
    </div>
  ),
}));

import StartOffsetBlock from './StartOffsetBlock';

describe('test StartOffsetBlock', () => {
  test('should render correctly', () => {
    const onValueChange = jest.fn();
    const { container } = render(<StartOffsetBlock onValueChange={onValueChange} value={0} />);

    expect(container).toMatchSnapshot();

    fireEvent.click(container.querySelector('div.option-block'));
    expect(container).toMatchSnapshot();

    fireEvent.change(container.querySelector('input.unit-input'), { target: { value: 100 } });
    expect(onValueChange).toHaveBeenCalledTimes(1);
    expect(onValueChange).toHaveBeenNthCalledWith(1, 100);
    expect(container).toMatchSnapshot();
  });

  test('should render correctly in mobile', () => {
    useIsMobile.mockReturnValue(true);

    const onValueChange = jest.fn();
    const { container } = render(<StartOffsetBlock onValueChange={onValueChange} value={0} />);

    expect(container).toMatchSnapshot();
  });
});
