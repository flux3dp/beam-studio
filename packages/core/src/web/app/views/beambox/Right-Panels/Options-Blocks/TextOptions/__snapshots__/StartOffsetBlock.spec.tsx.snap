// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test StartOffsetBlock should render correctly 1`] = `
<div>
  <div
    class="option-block"
  >
    <div
      class="label"
    >
      Text Offset
    </div>
    <div>
      mock-unit-input min:
      0
      max:
      100
      decimal:
      0
      defaultValue:
      0
      className:
      {"option-input":true}
      <input
        class="unit-input"
      />
    </div>
  </div>
</div>
`;

exports[`test StartOffsetBlock should render correctly 2`] = `
<div>
  <div
    class="option-block"
  >
    <div
      class="label"
    >
      Text Offset
    </div>
    <div>
      mock-unit-input min:
      0
      max:
      100
      decimal:
      0
      defaultValue:
      0
      className:
      {"option-input":true}
      <input
        class="unit-input"
      />
    </div>
  </div>
</div>
`;

exports[`test StartOffsetBlock should render correctly 3`] = `
<div>
  <div
    class="option-block"
  >
    <div
      class="label"
    >
      Text Offset
    </div>
    <div>
      mock-unit-input min:
      0
      max:
      100
      decimal:
      0
      defaultValue:
      0
      className:
      {"option-input":true}
      <input
        class="unit-input"
      />
    </div>
  </div>
</div>
`;

exports[`test StartOffsetBlock should render correctly in mobile 1`] = `
<div>
  <div>
    mock-number-item id:
    start_offset
    label:
    Text Offset
    min:
    0
    max:
    100
    decimal:
    0
    value:
    0
    <input
      class="unit-input"
    />
  </div>
</div>
`;
